staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -0,0 +1,211 @@
+			Groups: []string{user.AllUnauthenticated},
+		}))
+		apfHandler.ServeHTTP(w, r)
+	}), requestInfoFactory)
+
+	apfServer := httptest.NewServer(handler)
+	return apfServer
+}
+
+func TestApfSkipLongRunningRequest(t *testing.T) {
+	epmetrics.Register()
+
+	server := newApfServer(decisionError)
+	defer server.Close()
+
+	if err := expectHTTPGet(fmt.Sprintf("%s/api/v1/namespaces?watch=true", server.URL), http.StatusOK); err != nil {
+		// request should not be rejected
+		t.Error(err)
+	}
+
+	checkForExpectedMetrics(t, []string{
+		"apiserver_current_inflight_requests",
+	})
+}
+
+func TestApfRejectRequest(t *testing.T) {
+	epmetrics.Register()
+
+	server := newApfServer(decisionReject)
+	defer server.Close()
+
+	if err := expectHTTPGet(fmt.Sprintf("%s/api/v1/namespaces/default", server.URL), http.StatusTooManyRequests); err != nil {
+		t.Error(err)
+	}
+
+	// wait metrics to be collected
+	time.Sleep(time.Millisecond * 1500)
+
+	checkForExpectedMetrics(t, []string{
+		"apiserver_request_terminations_total",
+		"apiserver_dropped_requests_total",
+	})
+}
+
+func TestApfExemptRequest(t *testing.T) {
+	epmetrics.Register()
+	fcmetrics.Register()
+
+	// wait ReadWriteConcurrencyObserverPairGenerator to be registered
+	time.Sleep(time.Millisecond * 50)
+
+	server := newApfServer(decisionExempt)
+	defer server.Close()
+
+	if err := expectHTTPGet(fmt.Sprintf("%s/api/v1/namespaces/default", server.URL), http.StatusOK); err != nil {
+		t.Error(err)
+	}
+
+	// wait metrics to be collected
+	time.Sleep(time.Millisecond * 1500)
+
+	checkForExpectedMetrics(t, []string{
+		"apiserver_current_inflight_requests",
+		"apiserver_flowcontrol_read_vs_write_request_count_watermarks",
+		"apiserver_flowcontrol_read_vs_write_request_count_samples",
+	})
+}
+
+func TestApfExecuteRequest(t *testing.T) {
+	epmetrics.Register()
+	fcmetrics.Register()
+
+	// wait ReadWriteConcurrencyObserverPairGenerator to be registered
+	time.Sleep(time.Millisecond * 50)
+
+	server := newApfServer(decisionExecute)
+	defer server.Close()
+
+	if err := expectHTTPGet(fmt.Sprintf("%s/api/v1/namespaces/default", server.URL), http.StatusOK); err != nil {
+		t.Error(err)
+	}
+
+	// wait metrics to be collected
+	time.Sleep(time.Millisecond * 1500)
+
+	checkForExpectedMetrics(t, []string{
+		"apiserver_current_inflight_requests",
+		"apiserver_current_inqueue_requests",
+		"apiserver_flowcontrol_read_vs_write_request_count_watermarks",
+		"apiserver_flowcontrol_read_vs_write_request_count_samples",
+	})
+}
+
+func TestApfCancelWaitRequest(t *testing.T) {
+	epmetrics.Register()
+
+	server := newApfServer(decisionCancelWait)
+	defer server.Close()
+
+	if err := expectHTTPGet(fmt.Sprintf("%s/api/v1/namespaces/default", server.URL), http.StatusTooManyRequests); err != nil {
+		t.Error(err)
+	}
+
+	// wait metrics to be collected
+	time.Sleep(time.Millisecond * 1500)
+
+	checkForExpectedMetrics(t, []string{
+		"apiserver_current_inqueue_requests",
```

Comment:
This one also fails to show up in commit 3121fe16a20e046d85419a4aff06edbe220b39b2:
```
--- FAIL: TestApfCancelWaitRequest (1.51s)
    priority-and-fairness_test.go:208: Scraped metrics did not include expected metric apiserver_current_inqueue_requests
```

Reply:
Could you help check if it fail again? If so, could you tell me how did you reproduce it?

Reply:
I just tested with commit 4981cd1ae0812bdb783e70b03402b512fa73ba4c , and got no failures.

I test on my Mac by simply `cd` to the `filters` directory and then `go test` or, to be more specific, `go test -run TestApfSkipLongRunningRequest`.